src/util.ts

```diff
@@ -449,3 +449,34 @@ export function exec(command: string, args?: ReadonlyArray<string>, options?: cp
     });
     return disposable;
 }
+
+/**
+ * Check if an R package is available or not
+ *
+ * @param name the R package name that need to be checked
+ * @returns a boolean Promise
+ */
+export async function isRPkgIntalled(name: string, msg?: string): Promise<boolean> {
+     // users may write stdout in .Rprofile at the begining or the end of the session. so we need some strings to surround the output
+    const cmd = `cat(sprintf('vscodeRCheckPkgAvailability={%s}', require(${name}, quietly=TRUE)))`;
+    const rOut = /vscodeRCheckPkgAvailability=\{(.{4,5})\}/.exec(await executeRCommand(cmd));
+    if (rOut.length !== 2) {
+        // somehow the code failed to perform, so we don't throw (probably) false alert
+        return true;
+    }
+    const isInstalled = rOut[1] === 'TRUE';
+    if (!isInstalled && msg !== undefined) {
+        const repo = await getCranUrl();
+        const rPath = await getRpath(false);
+        void vscode.window.showErrorMessage(msg, 'Click to install', 'Not sure')
+        .then(function(select) {
+            if (select === 'Click to install') {
+                const args = [`--silent`, '--slave', `-e`, `install.packages('${name}', repos='${repo}')`];
```

Comment:
It seems that only around `--slave` are single quotes, not back quotes.

Reply:
Fixed, despite the code still works, but they should be the single quotes.